nestle.py

```diff
@@ -583,25 +619,58 @@ def set_options(self, options):
         self.enlarge = options.get('enlarge', 1.2)
 
     def update(self, pointvol):
+        self.cancel_queue()
         self.ells = bounding_ellipsoids(self.points, pointvol=pointvol)
         for ell in self.ells:
             ell.scale_to_vol(ell.vol * self.enlarge)
+        self.fill_queue()
+
+    def suggest_point(self):
+        while True:
+            u = sample_ellipsoids(self.ells, rstate=self.rstate)
+            if np.all(u > 0.) and np.all(u < 1.):
+                break
+        return u
 
     def new_point(self, loglstar):
         ncall = 0
         logl = -float('inf')
         while logl < loglstar:
-            while True:
-                u = sample_ellipsoids(self.ells, rstate=self.rstate)
-                if np.all(u > 0.) and np.all(u < 1.):
-                    break
-            v = self.prior_transform(u)
-            logl = self.loglikelihood(v)
+            u, v, logl = self.get_point_value()
             ncall += 1
 
         return u, v, logl, ncall
 
 
+# Do not derive from concurrent.futures.Executor because
+# concurrent.futures might not be available. We don't
+# need much functionality anyway.
+class FakePool(object):
+    def __init__(self):
+        pass
+
+    def submit(self, fn, *args, **kwargs):
+        return FakeFuture(fn, *args, **kwargs)
+
+    def map(self, func, *iterables, timeout=None, chunksize=1):
```

Comment:
Looks like this is a syntax error on Legacy Python. Change to `map(self, func, *iterables, **kwargs)`?